### ISSUE TITLE
Make Quant-LLM compatible with BF16

### ISSUE THREAD

**Issue by gau-nernst** [pytorch/ao#998]
Quant-LLM code: https://github.com/pytorch/ao/tree/main/torchao/csrc/cuda/fp6_llm

Currently Quant-LLM kernel (backing FPx in torchao) only works with FP16. This creates a small divergence from other quantization methods, which all work with BF16. Since all recent models are trained and released with BF16, having BF16 support potentially improve accuracy for FPx models.

Might be over-simplifying, but I think it's just the matter of modifying dequant logic and MMA instructions (as well as update dtype in function signature appropriately)

https://github.com/pytorch/ao/blob/09b8b3c3156b99fa8356e2df30c1a65cbbf5cc91/torchao/csrc/cuda/fp6_llm/utils_parallel_dequant.cuh#L30-L45

https://github.com/pytorch/ao/blob/09b8b3c3156b99fa8356e2df30c1a65cbbf5cc91/torchao/csrc/cuda/fp6_llm/ptx_mma.cuh#L117-L125

cc @msaroufim @HDCharles 

I might try to do it myself, but I think it would be an interesting good first issue task too. @tobiasvanderwerff Would you be interested?

**Comment by tobiasvanderwerff** [comment]
I'd love to work on this @gau-nernst :)

So if you're ok with me doing this I'll probably get started on it in the next few days.

**Comment by DevyRuxpin** [comment]
Please assign, I may have a solution for this.

**Comment by tobiasvanderwerff** [comment]
Hi @DevyRuxpin, I can understand that you want to work on this, but I have already made most of the necessary changes for this feature (see the diff [here](https://github.com/pytorch/ao/compare/main...tobiasvanderwerff:ao:fp6-bf16)). Since your solution is not complete yet (it removes FP16 support), it's probably best to avoid doing further duplicate work.
